### PR TITLE
[WIP] Multi Task Learning

### PR DESCRIPTION
**Pull request by chidauri**
#40 
**Currently implemented for bugtype model only**

### CODE REVIEW

**Comment by ayush-1506** [comment]
@chidauri Why are you unable to run this on your system? I'd love to see how this compares to our current bugtype model. You could try running on colab, or reducing the size of the dataset you're training on.

**Comment by chidauri** [comment]
@ayush-1506 @marco-c I am only able to run it for 7000 bugs , if i go more, memory overflow occurs :(
**For 7000 bugs:**
```
6300/6300 [==============================] - 21s 3ms/step - loss: 0.1978 - acc: 0.9311
Epoch 2/2
6300/6300 [==============================] - 12s 2ms/step - loss: 0.1066 - acc: 0.9769
```

**Comment by ayush-1506** [comment]
I'm no expert, but an accuracy of 0.97 sounds too good to be true.

**Comment by chidauri** [comment]
> I'm no expert, but an accuracy of 0.97 sounds too good to be true.

yeah it is probably overfitting(probably due to such a small training set ), so first of all we need to focus on reducing memory requirements so that we can train on larger dataset , i tried component_nn.py , that model suffers from the same issue

**Comment by ayush-1506** [comment]
I was able to run models on the entire DB on colab (though they weren't deep learning models). I guess you could wait for Marco to return so he could test it on the 64 gb RAM cluster he has.

**Review by marco-c — CHANGES_REQUESTED** [review]
Will need to be rebased but I will keep this open in case we want to pick it up again.
